### PR TITLE
v1 breaking changes

### PR DESCRIPTION
**Pull request by joellabes**
This is a:
- [ ] bug fix PR with no breaking changes — please ensure the base branch is `main`
- [ ] new functionality — please ensure the base branch is the latest `dev/` branch
- [x] a breaking change — please ensure the base branch is the latest `dev/` branch

## Description & motivation
- Change `require-dbt-version` to `[">=1.0.0", "<2.0.0"]`
- Bump dbt-utils dependency
- Replace `source-paths` and `data-paths` with `model-paths` and `seed-paths` respectively
- Rename `data` and `analysis` directories to `seeds` and `analyses` respectively
- Replace `dbt_modules` with `dbt_packages` in `clean-targets`
- Restore `--warn-error` to CI config
- Remove redundant run steps from CI (there are no models, so combined with `warn-error` it was failing. i don't know how this ever worked)

## Checklist
- [x] I have verified that these changes work locally
- [x] I have updated the README.md (if applicable)
- [x] I have added tests & descriptions to my models (and macros if applicable)
- [x] I have added an entry to CHANGELOG.md

### CODE REVIEW

**Review by github-christophe-oudar — APPROVED** [review]
LGTM, looking forward to this release as it prevents bumping to projects with dbt 1.0.0, dbt-labs/dbt_utils 0.8 and dbt-labs/codegen 🙌

File: dbt_project.yml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 name: 'codegen'
 version: '0.4.0'
```
[the remark this answers is not in the document]
Comment (joellabes) [reply dbt_project.yml:2]: Good spotting! This value has no semantic meaning, but worth fixing anyway.

**Comment by joellabes** [comment]
@github-christophe-oudar, there's also a 0.7.6 release of utils which is compatible with the currently shipping version of codegen and dbt 1.0. You can use that without errors in the meantime! Hopefully we'll get this merged soon though

**Comment by github-christophe-oudar** [comment]
Sure, I just wanted to bump both libs altogether and ran into this one 👋
Thanks!

**Review by jasnonaz — APPROVED** [review]
very good, very good, very good as always Mr. Labes

**Pull request merged**